tests/bluetooth/blob_io_flash/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+tests:
+  blob_io_flash.flash_io:
+    platform_allow: nrf51dk_nrf51422 qemu_x86 native_posix native_posix_64
```

Comment:
nrf51 is a deprecated platform with low capabilities. No needs to add for the tests. If you need to run it on real hardware please use `nrf52840dk_nrf52840`.

Did you run this test on all listed platforms?

Reply:
I see now that there is a problem on qemu, I'll test the rest too

Reply:
I'm removing targets other than native, because I use arrays with test data of very big sizes to check BLOB data written and read from flash. Testing this on ARM platforms, including QEMU, could be done in the future as BSIM test.

Reply:
Then we can consider that data verification is covered here and module running in real BLOB transfer will happen on simulated HW

Reply:
I do not think it is worth to have two tests (unit and bsim) for the same functionality. This is overcomplication. Did you try to run it on nrf52840?

Reply:
Because I'm using big arrays to verify what was written into/read from flash it's not possible to use it on any real HW or qemu, because of flash limitation. And I think it's necessary to assure that correct data is stored in flash - that's how I found bug in `block_start`.

One idea is to limit flash area size so much that it would fit on stack - that way I can write whole partition and read it out to compare with control data. I suspect this could be done with `.dts` overlay, but I'm not sure. I don't want to just use data smaller than partition: I want to stress test this, write many chunks with maximum size, reach end of flash, write more and see if error triggers. IIRC partition size is something around 43kB, no real board has RAM that big. But this test is not running on embedded, so using native_posix I can afford making arrays this big

Reply:
I'm Ok to run it only on native_posix platform. This is the unit test. The platform really doesn't matter. I worry the only thing there shouldn't be platforms that do not work with test.